packages/btcindexer/src/btcblock.ts

```diff
@@ -1,19 +1,49 @@
-export interface Tx {
-	id: string;
+import { Block } from 'bitcoinjs-lib';
 
-	raw: Uint8Array; // raw tx bytes
+export interface ExtendedBlock extends Block {
+	raw: Buffer;
 }
 
-export interface Block {
-	id: string;
-	txs: Tx[];
+export async function parseBlocks(body: ReadableStream | null): Promise<ExtendedBlock[]> {
+	if (!body) {
+		return [];
+	}
 
-	// should not be serialized to JSON
-	raw: Uint8Array; // raw block bytes
-}
+	const reader = body.getReader();
+	const chunks: Uint8Array[] = [];
+	let totalLen = 0;
+	while (true) {
+		const { done, value } = await reader.read();
+		if (done) break;
+		chunks.push(value);
+		totalLen += value.length;
+	}
+
+	const fullBuffer = new Uint8Array(totalLen);
+	let bufferOffset = 0;
+	for (const chunk of chunks) {
+		fullBuffer.set(chunk, bufferOffset);
+		bufferOffset += chunk.length;
+	}
 
-export function parseBlocks(body: ReadableStream | null): Block[] {
-	// TODO: use msgPack or the default Bitcoin serializer
-	console.log('parsing blocks', body);
-	return [];
+	const nodeBuffer = Buffer.from(fullBuffer);
+
+	const blocks: ExtendedBlock[] = [];
+	let parseOffset = 0;
+	while (parseOffset < nodeBuffer.length) {
```

Comment:
Assume: 
- We can parse the first, second, third... block in the data. 
- The fourth block is invalid. 
Should we return first, second, third, or return an error?